src/GafferOSL/OSLExpressionEngine.cpp

```diff
@@ -323,7 +323,21 @@ class OSLExpressionEngine : public Gaffer::Expression::Engine
 		    OSL::ShaderGlobals shaderGlobals;
 			memset( &shaderGlobals, 0, sizeof( ShaderGlobals ) );
 
-			shaderGlobals.time = context->getTime();
+			int numGlobals = 0;
```

Comment:
This is repeating work needlessly for every single expression execution. Can we just add a `bool m_needsTime` and set it at line 302 in `parse()`? Then this becomes simply `if( m_needsTime ) { shaderGlobals.time = context->getTime(); }`.

Reply:
Note that there is no binary compatibility issue with adding a data member because the OSLExpressionEngine class isn't public.

Reply:
Addressed in baba31b5218dbaec2e799005ee5d986fb65e8b56